lib/jbuilder/collection_renderer.rb

```diff
@@ -0,0 +1,118 @@
+require 'delegate'
+require 'active_support/concern'
+
+begin
+  require 'action_view/renderer/collection_renderer'
+rescue LoadError
+  require 'action_view/renderer/partial_renderer'
+end
+
+class Jbuilder
+  module CollectionRenderable # :nodoc:
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def supports_collection_rendering?
+        superclass.private_method_defined?(:build_rendered_template) && superclass.private_method_defined?(:build_rendered_collection)
+      end
+    end
+
+    private
+
+    def build_rendered_template(content, template, layout = nil)
+      super(content || json.attributes!, template)
+    end
+
+    def build_rendered_collection(templates, _spacer)
+      json.merge!(templates.map(&:body))
+    end
+
+    def json
+      @options[:locals].fetch(:json)
+    end
+
+    class ScopedIterator < ::SimpleDelegator # :nodoc:
+      include Enumerable
+
+      def initialize(obj, scope)
+        super(obj)
+        @scope = scope
+      end
+
+      # Rails 6.0 support:
+      def each
+        return enum_for(:each) unless block_given?
+
+        __getobj__.each do |object|
+          @scope.call { yield(object) }
+        end
+      end
+
+      # Rails 6.1 support:
+      def each_with_info
+        return enum_for(:each_with_info) unless block_given?
+
+        __getobj__.each_with_info do |object, (path, as, counter, iteration)|
+          @scope.call { yield(object, [path, as, counter, iteration]) }
+        end
+      end
+    end
+
+    private_constant :ScopedIterator
+  end
+
+  collection_renderer_available = begin
+                                     ::ActionView::CollectionRenderer
+
+                                     true
+                                   rescue ::NameError
+                                     false
+                                   end
+
+  if collection_renderer_available
+    # Rails 6.1 support:
+    class CollectionRenderer < ::ActionView::CollectionRenderer # :nodoc:
+      include CollectionRenderable
+
+      def initialize(lookup_context, options, &scope)
+        super(lookup_context, options)
+        @scope = scope
+      end
+
+      private
+
+        def collection_with_template(view, template, layout, collection)
+          super(view, template, layout, ScopedIterator.new(collection, @scope))
+        end
+    end
+  else
+    # Rails 6.0 support:
+    class CollectionRenderer < ::ActionView::PartialRenderer # :nodoc:
+      include CollectionRenderable
+
+      def initialize(lookup_context, options, &scope)
+        super(lookup_context)
+        @options = options
+        @scope = scope
+      end
+
+      def render_collection_with_partial(collection, partial, context, block)
+        render(context, @options.merge(collection: collection, partial: partial), block)
```

Comment:
Can we omit the `block` since we pass `nil` in the rendering?

Reply:
Or I'm guessing that's for Rails 6.1?

Reply:
Correct, I'm basically following [6.1's interface](https://github.com/rails/rails/blob/6efe2b490083c90550786ab70957e8dc548baf23/actionview/lib/action_view/renderer/collection_renderer.rb#L104) and eventually remove this method once support for 6.0 <= is dropped.